cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -176,54 +179,75 @@ func (sub EthereumSub) Start(completionEvent *sync.WaitGroup) {
 	}
 	defer subHead.Unsubscribe()
 
+	lastProcessedBlock := big.NewInt(0)
```

Comment:
We also talked about writing out some local data on where the relayer is, so it could just read that on restart if it were available.

Reply:
@banshee on-chain query would be more robust, and remove the risk of local becoming out of sync with onchain for whatever reason

Reply:
Totally agree that on-chain is better than local.  Local was just something we discussed and didn't implement, and has lots of obvious drawbacks (what if you loose the machine, for example).

Reply:
I agree that this would be a really great feature to add in the future. For now, I think that instead of starting at block 0, it should just start at the block number that is passed in from an environment variable. This way, if we need to restart, it doesn't give people extra money on the sifchain side that they shouldn't have.